packages/flutter_test/lib/src/binding.dart

```diff
@@ -848,6 +854,13 @@ class AutomatedTestWidgetsFlutterBinding extends TestWidgetsFlutterBinding {
     });
 
     return new Future<Null>.microtask(() async {
+      // Run all queued microtasks.
+      await new Future<Null>.microtask(() {});
```

Comment:
I don't understand the purpose of this initial delay. This whole function is already run in a microtask.

Reply:
But all the microtasks in _runTest are run in the FakeAsync zone, no? What actual microtasks get scheduled in this zone? The whole point of this loop is to flush the microtasks, which we do with an await in the while loop; why does pushing us to the end of the event loop help here?

Reply:
Ok. I looked at this in more detail.

So the comment in line 875 was wrong.
It stated that by returning the `testBodyResult` future from a future that was created outside the `fakeAsync` it would not need the `fakeAsync` anymore.
That's not true. The `testBodyResult` future might still get a `.then` call. It just happened not to be the case, because we have an optimization that propagates results without explicitly invoking `.then` if we know already the next listener.

By delaying twice the next listener was already known. By delaying only once, an internal `.then` call was added to the `testBodyResult`. However, since there was no additional `fakeAsync.fushMicrotask` call that `.then` call was never executed.

I refactored the code.

Reply:
ah, this makes sense.